src/tools.rs

```diff
@@ -591,16 +592,16 @@ pub fn remove_subject_prefix(last_subject: &str) -> String {
 
 // Types and methods to create hop-info for message-info
 
-fn extract_address_from_receive_header<'a>(header: &'a str, start: &str) -> Option<&'a str> {
-    let header_len = header.len();
-    header.find(start).and_then(|mut begin| {
-        begin += start.len();
-        let end = header
-            .get(begin..)?
-            .find(|c: char| c.is_whitespace())
-            .unwrap_or(header_len);
-        header.get(begin..begin + end)
-    })
+fn extract_address_from_receive_header(header: &str, start: &str) -> Option<String> {
+    let c: inline_python::Context = python! {
+        header_len = len('header)
```

Comment:
In case you're wondering, the `'` before `header` tells inline_python that this is a Rust variable I want to use inside Python.